src/bdecode.cpp

```diff
@@ -477,22 +477,22 @@ namespace libtorrent
 
 		while (tokens[token].type != bdecode_token::end)
 		{
-			bdecode_token const& t = tokens[token];
-			TORRENT_ASSERT(t.type == bdecode_token::string);
-			int const size = m_root_tokens[token + 1].offset - t.offset - t.start_offset();
+			bdecode_token const* t = &tokens[token];
+			TORRENT_ASSERT(t->type == bdecode_token::string);
+			int const size = (t + 1)->offset - t->offset - t->start_offset();
```

Comment:
"m_root_tokens" reference instead of local "tokens" comes from original "bdecode land" commit. from "bdecode_node::string_length()" implementation i figure out that "tokens" should definilty be used to access next item.